packages/lib-classifier/src/store/TutorialStore.js

```diff
@@ -56,20 +63,21 @@ const TutorialStore = types
     },
 
     get hasNotSeenTutorialBefore () {
-      const upp = getRoot(self).userProjectPreferences.active
+      const validUPPReference = isValidReference(() => getRoot(self).userProjectPreferences.active)
```

Comment:
I've been using [`tryReference()`](https://github.com/mobxjs/mobx-state-tree#reference-validation-isvalidreference-tryreference-oninvalidated-hook-and-typessafereference) in recent code, which returns a valid tree node or undefined. It's useful when you want to set a variable that refers to a valid tree node.
```js
const upp = tryReference(() => getRoot(self).userProjectPreferences.active
```

Reply:
I keep forgetting about this API method. I'll have a look at it. Thanks.